src/main/java/org/llorllale/cactoos/matchers/IsMatcherOf.java

```diff
@@ -0,0 +1,180 @@
+     */
+    private final String exp;
+
+    /**
+     * The actual message with testing details from target matcher.
+     */
+    private final String act;
+
+    /**
+     * The testing arguments for the target matcher.
+     */
+    private final X margs;
+
+    /**
+     * Ctor.
+     * @param status The expected testing status within matcher.
+     * @param margs The testing arguments for the matcher.
+     */
+    public IsMatcherOf(final boolean status, final X margs) {
+        this(status, margs, "");
+    }
+
+    /**
+     * Ctor.
+     * @param status The expected testing status within matcher.
+     * @param margs The testing arguments for the matcher.
+     * @param act The actual message with testing details from target matcher.
+     */
+    public IsMatcherOf(final boolean status, final X margs, final String act) {
+        this(status, margs, act, "");
+    }
+
+    /**
+     * Ctor.
+     * @param status The expected testing status within matcher.
+     * @param margs The testing arguments for the matcher.
+     * @param act The actual message with testing details from target matcher.
+     * @param exp The expected message with testing details from target matcher.
+     * @checkstyle ParameterNumberCheck (5 lines)
+     */
+    public IsMatcherOf(final boolean status, final X margs, final String act,
+        final String exp) {
+        super();
+        this.status = status;
+        this.margs = margs;
+        this.act = act;
+        this.exp = exp;
+    }
+
+    @Override
+    public void describeTo(final Description desc) {
+        this.describe(this.status, this.exp, this.act, desc);
+    }
+
+    @Override
+    protected boolean matchesSafely(final Matcher<X> mtcr,
+        final Description dsc) {
+        final Description expected = new StringDescription();
+        final Description actual = new StringDescription();
+        mtcr.describeMismatch(this.margs, actual);
+        mtcr.describeTo(expected);
+        final boolean matched;
+        if (this.exp.trim().isEmpty() || this.act.trim().isEmpty()) {
+            matched = this.status == mtcr.matches(this.margs);
+            this.describe(matched, "", "", dsc);
+        } else {
+            matched = this.status == mtcr.matches(this.margs)
+                && this.act.equals(actual.toString())
+                && this.exp.equals(expected.toString());
+            this.describe(matched, actual.toString(), expected.toString(), dsc);
+        }
+        return matched;
+    }
+
+    /**
+     * Describe testing result of target matcher in hamcrest terms.
+     * @param sttus The expected testing status within matcher
+     * @param ectd The expected message with testing details from target matcher
+     * @param actl The actual message with testing details from target matcher
+     * @param desc The hamcrest description of the testing scenario
+     * @checkstyle ParameterNumberCheck (5 lines)
+     * @checkstyle NonStaticMethodCheck (5 lines)
```

Comment:
@dgroup let's just make `describe` a static method and remove this checkstyle-ignore